### ISSUE TITLE
Minimal os versions?

### ISSUE THREAD

**Issue by matopeto** [Qata/MonadicJSON#4]
In Packgate.swift you have:

```
        .macOS(.v12), .iOS(.v15), .watchOS(.v8), .tvOS(.v15),
```

Is really the latest OSs needed for this package? Can be platforms lowered to lowest possible OS that code supports?

**Comment by matopeto** [comment]
When i try to compule for e.g. ios 13 i got two incompatible errors:

```
'some' return types are only available in iOS 13.0.0 or newer
```

and 

```
'init(_:format:lenient:)' is only available in iOS 15.0 or newer
```
on row:
```
                return try .init(Decimal(value, format: .number))
```


I think you could lower the requirements of iOS version at least to ios13 by using compatible Decimal initializer.

It would be nice to use this json decoder in older projects (we still supports iOS12, but if this project will supports at least ios13 it would be very nice... iOS 15 is too high requirements

**Comment by Qata** [comment]
I've just pushed up a version with minimum targets as low as they can be set.
